tests/WatcherFactoryTest.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace Spatie\PhpUnitWatcher\Test;
+
+use PHPUnit\Framework\TestCase;
+use Spatie\PhpUnitWatcher\WatcherFactory;
+
+class WatcherFactorTest extends TestCase
+{
+	public function setUp() {
+		parent::setUp();
+
+		// todo need to create foo and bar folders inside current working dir
+	}
+
+	public function tearDown() {
+		parent::tearDown();
+
+		// todo revert setUp();
+	}
+
+	/** @test */
+    public function it_can_be_instantiated()
+    {
+        $factory = new WatcherFactory();
+
+        $this->assertInstanceOf(WatcherFactory::class, $factory);
+    }
+
+    /**
+     * @test
+     *
+     * @covers WatcherFactory::create
+     *
+     * @dataProvider data_options_are_replaced_recursively
+     */
+	public function options_are_replaced_recursively( $userOptions, $expectedOptions ) {
+		$watcher = WatcherFactory::create( $userOptions );
+		$actualOptions = $watcher[1];
+
+		$this->assertEquals( $expectedOptions, $actualOptions );
+	}
+
+	public function data_options_are_replaced_recursively() {
```

Comment:
This is named like a test. Rename this to `getOptions`. That makes it more clear that it is a dataProvider.

Reply:
I ended up taking a different approach to get the tests working in ae8875a6fc6fc44858662669ce33adaf28750892, so there's no longer a data provider.